src/OpenTelemetry.Api/Trace/TracerProviderBuilder.cs

```diff
@@ -39,8 +39,8 @@ internal TracerProviderBuilder()
         /// <summary>
         /// Adds given activitysource names to the list of subscribed sources.
         /// </summary>
-        /// <param name="names">Activity source names.</param>
+        /// <param name="sources">Activity source names.</param>
         /// <returns>Returns <see cref="TracerProviderBuilder"/> for chaining.</returns>
-        public abstract TracerProviderBuilder AddSource(params string[] names);
+        public abstract TracerProviderBuilder AddSource(params Source[] sources);
```

Comment:
We can keep the string version as well, and internally convert that to use Source. Most users don't care about Version, so they can use the easy version.

Reply:
ok. i will create a second method with sources then. THanks for the feedback!

Reply:
basically retaining the current API.
the version support is an optional enhancement.

Reply:
just added again and reverted some changes :) Let me know what do you think